src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -1233,7 +1237,22 @@ private void VisitVariableNode(ParseNode node)
             identifierIsVariable = false;
         }
 
-        private void VisitExpression(ParseNode node)
+        /// <summary>
+        /// Performs the work for a number of different expressions that all
+        /// share the following universal basic properties:<br/>
+        /// - They contain optional binary operators.<br/>
+        /// - The terms are all at the same precedence level.<br/>
+        /// - Because of the tie of precedence level, the terms are to be evaluated left-to-right.<br/>
+        /// - No special extra work is needed, such that simply doing "push expr1, push expr2, then do operator" is all that's needed.<br/>
+        /// <br/>
+        /// Examples:<br/>
+        ///   5 + 4 - x + 2 // because + and - are in the same parse rule, these all get the same flat precedence.<br/>
+        ///   x * y * z<br/>
+        /// In cases like that where all the operators "tie", the entire chain of terms lives in the same ParseNode,<br/>
+        /// and we have to unroll those terms and presume left-to-right precedence.  That is what this method does.<br/>
+        /// </summary>
+        /// <param name="node"></param>
+        private void VisitExpressionChain(ParseNode node)
```

Comment:
Note, I didn't actually change anything in this method at all, other than to add the documentation comment.  This is the method that *previously* had been used for AND and OR expressions, along with all the binary operators for math - they were handled by the same method.  As I looked at this and started to edit it, I came to the conclusion that the code was getting too messy when I intermingled it with this method, so I split it out into a new method, VisitShortCircuitBoolean().  But I figured it didn't hurt to leave the explanatory comments behind because previously the method wasn't really explaining itself well.